### ISSUE TITLE
Can't evaluate the pretrained adapters

### ISSUE THREAD

**Issue by mahnerak** [Adapter-Hub/adapter-transformers#113]
I'm having trouble evaluating the pretrained adapters from the AdapterHub.
I use the `run_glue.py` script:
```
python examples/text-classification/run_glue.py \
--do_eval \
--task_name sst2 \
--model_name_or_path roberta-large \
--load_adapter sentiment/sst-2@ukp \
--output_dir /tmp/adapter-out  \
--train_adapter
```

As a result, I get this output:
```
12/30/2020 20:03:57 - INFO - __main__ -   ***** Eval results sst2 *****
12/30/2020 20:03:57 - INFO - __main__ -     eval_loss = 0.6205388307571411
12/30/2020 20:03:57 - INFO - __main__ -     eval_accuracy = 0.6192660550458715
```
...Which is obviously not I expected.

The model page (https://adapterhub.ml/adapters/ukp/roberta-large-sst_houlsby/) states that it achieves:
> Achieves 96.33 Accuracy on the SST-2 devset

**Comment by mahnerak** [comment]
Also, it can easily take the `--load_adapter` while the adapter is not enabled with `--train_adapter` flag.
We may need some kind of warning/error here.

**Comment by calpt** [comment]
Hi @mahnerak,

The adapter you're trying to load was not trained using the model class from the example script, which results in the prediction head coming with the adapter not being loaded correctly. (We will try to make this clearer). In the `run_glue.py` script, try to change the class from `AutoModelForSequenceClassification` to `AutoModelWithHeads`. With this change, I can get the stated accuracy.

**Comment by mahnerak** [comment]
Thanks, @calpt, it worked!

But it seems like it may fail in some cases. For example, if I pick a `pfeiffer` adapter trained on MNLI:
```sh
python examples/text-classification/run_glue.py \
--fp16 --do_eval \
--task_name mnli \
--model_name_or_path roberta-large \
--load_adapter nli/multinli@ukp \
--output_dir /tmp/adapter-out  \
--train_adapter --adapter_config pfeiffer
```
It shows:
```
01/20/2021 14:17:06 - INFO - __main__ -   ***** Eval results mnli *****
01/20/2021 14:17:06 - INFO - __main__ -     eval_loss = 4.407608985900879
01/20/2021 14:17:06 - INFO - __main__ -     eval_accuracy = 0.29077941925624046

01/20/2021 14:17:48 - INFO - __main__ -   ***** Eval results mnli-mm *****
01/20/2021 14:17:48 - INFO - __main__ -     eval_loss = 4.434850692749023
01/20/2021 14:17:48 - INFO - __main__ -     eval_accuracy = 0.28834418226200165
```

Is there another modification I need to apply?

**Comment by calpt** [comment]
@mahnerak Your results sounded suspicious to me as only the results for MNLI were very low when I checked again, so I did some further investigation. Apparently, there was a breaking change in the GLUE training script between Hugginface transformers v2.11 (with which most of the adapters on the Hub are trained) and the current v3.5.1 that has gone unnoticed by us: **The order of class labels has changed.**

v2.11 ([source](https://github.com/huggingface/transformers/blob/b42586ea560a20dcadb78472a6b4596f579e9043/src/transformers/data/processors/glue.py#L231-L233)) has this mapping: `contradiction -> 0; entailment -> 1; neutral -> 2`

v3.5.1 (now using Huggingface datasets, [source](https://github.com/huggingface/datasets/blob/dbead924acd1fce76fd0328327728ea333da99a7/datasets/glue/glue.py#L56)) has the mapping `entailment -> 0; neutral -> 1; contradiction -> 2`

I swapped the ids of the adapter predictions in the evaluation method (0 -> 2, 1 -> 0, 2 -> 1) and I'm getting the stated accuracy again.
